frontend/__tests__/Login.test.js

```diff
@@ -0,0 +1,114 @@
+import '@testing-library/jest-dom'
+import { render, screen, fireEvent, waitFor } from '@testing-library/react'
+import { act } from 'react-dom/test-utils'
+import { MemoryRouter } from 'react-router-dom'
+import axios from 'axios'
+import Login from '../src/pages/Login'
+
+jest.mock('../src/assets/logo.png', () => './__mocks__/logoMock.js')
+jest.mock('axios')
+
+describe('Login Component', () => {
+  // TEST #1
+  test('renders Login component', () => {
+    render(<MemoryRouter><Login /></MemoryRouter>)
+  })
+
+  // TEST #2
+  test('renders Login component with logo', () => {
+    render(<MemoryRouter><Login /></MemoryRouter>)
+    const logoElement = screen.getByRole('img', { name: '' })
+    expect(logoElement).toBeInTheDocument()
+    expect(logoElement).toHaveClass('size-1/2 object-contain')
+    expect(logoElement).toHaveAttribute('src', './__mocks__/logoMock.js')
+  })
+
+  // TEST #3
+  test('renders login form with input fields', () => {
+    render(<MemoryRouter><Login /></MemoryRouter>)
+    const usernameInput = screen.getByPlaceholderText('Username')
+    const passwordInput = screen.getByPlaceholderText('Password')
+    const loginButton = screen.getByRole('button', { name: 'Login' })
+
+    expect(usernameInput).toBeInTheDocument()
```

Comment:
I am not sure about these assertions, its testing the rendering of the login form component which already has its own render tests. A better test could be rending the Login page with the image, LoginForm component mocked, and <p> then assert that all of those elements are present in the rendered Login page

Reply:
Also adding I think it should all be in 1 test. So a "good" Login page has to be rendered with all three- the Logo, <LoginForm />, and and the para. This should be 1 test that asserts that all 3 are present.